### ISSUE TITLE
test_icrs_consistency and test_against_pyephem failing with pyerfa 1.7.2

### ISSUE THREAD

**Issue by saimn** [astropy/astropy#11305]
This test is failing following the release of pyerfa 1.7.2 (it passes locally with 1.7.1 and fails with 1.7.2):
https://github.com/astropy/astropy/runs/1850026053?check_suite_focus=true#step:6:933
```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-06
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 3.15489777e-05
E       Max relative difference: inf
E        x: array(3.154898e-05)
E        y: array(0.)
```

# TODO

- [x] Revert #11308

**Comment by mhvk** [comment]
The new SOFA version wrapped in 1.7.2 has changes to the polar motion, making them more accurate.  I noticed that quite a few of the SOFA tests themselves were changed (and I know I copied some of those for `EarthLocation`, etc., to astropy). But then, it would seem this should *not* affect the consistency tests!

I guess the first question now is what to do: 
1. Fix erfa to 1.7.1 while we check?
2. Increase the tolerance slightly or xfail the test?

One lesson learned is that I really should have checked astropy with the new erfa...

**Comment by pllim** [comment]
To clarify: There are 201 tests failing because of this -- partial log at #11307

**Comment by mhvk** [comment]
Shall we just use `pyerfa < 1.7.2` for now? I can look into the failures in detail this only at the end of the day...

**Comment by olebole** [comment]
On Debian, we now also [see a failure](https://ci.debian.net/data/autopkgtest/testing/amd64/a/astropy/10386495/log.gz) -- however  only on `test_against_pyephem`. Do I understand it right that this is mainly a problem of the test, so that I could just disable it?

**Comment by pllim** [comment]
🤷 
```
        # Current value: 0.0031402822944751997 arcsec
>       assert distance < 1 * u.arcsec
E       assert <Angle 521.28428171 arcsec> < (1 * Unit("arcsec"))
E        +  where Unit("arcsec") = u.arcsec
```

**Comment by mhvk** [comment]
@olebole - yes, it is a problem with the test, which checks that results do not change relative to the older version of `erfa` - but the newer version does polar motion more accurately.

**Comment by mkbrewer** [comment]
@mhvk This should be an easy fix. You just need to update ErfaAstrom.apio() to use the new polar motion algorithm.

**Comment by mhvk** [comment]
Yes, agreed the fix is obvious. It is a matter of getting around to it...

**Comment by pllim** [comment]
Looks like this was unresolved for 4.2.1. I am tentatively milestoning this issue to 4.3 so we don't forget about it before next major release.

**Comment by mhvk** [comment]
Finally got around to this. See #11485 for a fix that gets rid of all but one of the errors. Am not yet seeing the solution for the last one... Help appreciated!

**Comment by pllim** [comment]
Resolved by #11485. Thanks, everyone!
